### DISCUSSION TITLE
Ordo Cisterciensis: Vigilia Imm. Conceptionis: omnia ex Feria: how to do it?

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#4177] [Developers corner]
Laudétur Jesus Christus!

I have come to the Vigilia Immaculatæ Conceptionis B.M.V. on 7. Dec. (there is no S. Ambrose in the Cistercian Ordo this day), and the rubric says that this day, no Votive Office or 3 Lect. feast may be taken, only the Advent Feria. So the logical thing to do is to change the title to reflect the vigil, so that I don't ask myself, why no _B.M.V. in Sabbato_ or other similar thing is not selected, and the rest take from the Feria.

My question is: is there any way in the datafiles in `Sancti `folder, to take the office from the feria and ignore the BMVSabbato? 

One obvious solution comes to mind: ignore the datafiles, add an exception to the `BMVSabbato` code and change the title in the code as well. But I don't really like it....

Thanks for the answer, God Bless!

**Comment by FAJ-Munich** [comment]
Given the lowering of the rank of Advent for Cist to 1.15 should give way to the Vigil given a rank of 1.5 should do the trick. But it needs to be put in a separate file like `12-07C.txt`. I hope the routine suppressing vigils does not interfere with it. Otherwise for the time being, you have to misspell Vigil so that the routine doesn't catch it. 

Best thing to model it similar to other vigil data files. E.g. 10-31 or 06-23 should do.

**Reply by Augustinus-Altovadensis** [reply]
> Given the lowering of the rank of Advent for Cist to 1.15 should give way to the Vigil given a rank of 1.5 should do the trick. But it needs to be put in a separate file like `12-07C.txt`. I hope the routine suppressing vigils does not interfere with it. Otherwise for the time being, you have to misspell Vigil so that the routine doesn't catch it.
> 
This was the first thing I tried, but on my as well as your version, it simply takes the Vigil as a separate feast, it says `Ant 2 is missing`, and then it adds the commemoration of Feria. Which is not what is supposed to happen. Never mind, I just added an exception for Dec. 6. and 7. to `$BMVSabbato` and added a comment (`$dayname[2]`) under every weekday, that is not Sunday. Two commented lines of code and it solves the issue. 

> Best thing to model it similar to other vigil data files. E.g. 10-31 or 06-23 should do.
Yes. The thing is that none of them use the Sunday collect, which is also important, because it can be first or second Sunday in Advent... And yes, I used the `Oratio dominica` rule.

BTW, I have updated almost all the `Sancti` files in my fork, and I have updated the ranks to match your version. So you can use directly my files. 
I have one more question: does the current version of the program allow for different versicles with the `(rubrica cisterciensis)` magic comment? If so, I could start implementing the Sancti and Tempora files.
